ts/output/common/LineBBox.ts

```diff
@@ -0,0 +1,155 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2022 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements the LineBBox class
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+import {MmlNode} from '../../core/MmlTree/MmlNode.js';
+import {BBox, BBoxData} from '../../util/BBox.js';
+import {StringMap} from './Wrapper.js';
+
+/*****************************************************************/
+
+
+/**
+ * align and indent values
+ */
+export type IndentData = [string, string];
+
+/**
+ * A BBox with extra data about line breaking
+ */
+export class LineBBox extends BBox {
+
+  /**
+   * Indentation data for the line break
+   */
+  public indentData: IndentData[] = null;
+
+  /**
+   * the lineleading for this break
+   */
+  public lineLeading: number;
+
+  /**
+   * The original BBox.L value
+   */
+  public originalL: number;
+
+  /**
+   * True if this bbox is at the beginning of a line
+   */
+  public isFirst: boolean = false;
+
+  /**
+   * The index of the child node containing this starting line break and the line within that child
+   */
+  public start: [number, number];
+  /**
+   * The index of the child node containingt the ending line break and the line within that child
+   */
+  public end: [number, number];
+
+  /**
+   * @param {BBox} bbox           The bbox to extend
+   * @param {number} leading      The lineleading value for the break
+   * @param {IndentData} indent   The align/shift information
+   * @return {ExtendedBBox}       The bbox extended
+   */
+  public static from(bbox: BBox, leading: number, indent: IndentData[] = null): LineBBox {
+    const nbox = new this();
```

Comment:
Does that work? Is `this` not the object?

Reply:
Because this is a static method, `this` is the object class, so yes, this works.  It allows a subclass (if needed) of `LineBBox` to use the unmodified static method to create the new subclass as well as the original class.  I try not to use explicit class names unless absolutely necessary in order to make subclassing easier.

Reply:
Thanks for the explanation.